src/main/java/org/folio/circulation/domain/CheckInContext.java

```diff
@@ -182,7 +192,24 @@ public CheckInContext withItemStatusBeforeCheckIn(ItemStatus itemStatus) {
       this.loggedInUserId,
       this.checkInProcessedDateTime,
       this.inHouseUse,
-      itemStatus);
+      itemStatus,
+      this.timeZone);
+  }
+
+  public CheckInContext withTimeZone(ZoneId timeZone) {
```

Comment:
Can we use `@With` class annotation and also remove other `with...()` methods from this class? We usually do this whenever we need to add one more `with...()` method. Example - `LoanAndRelatedRecords` class

Reply:
Done